### PR TITLE
add test case to function pushImage

### PR DESCRIPTION
**Pull request by guangxuli**

### CODE REVIEW

File: pkg/build/builder/dockerutil_test.go [hunk 1]
```diff
@@ -108,3 +126,42 @@ func TestTagImage(t *testing.T) {
...
+	testImageName = "repo_foo_bar:tag_test_succ_foo_bar"
+	if err := pushImage(fakeDocker, testImageName, testAuth); err != nil {
+		t.Errorf("Unexpect push image : %v, want succ", err)
+	}
+
+	//expect fail, retry 2 times
+	testImageName = "repo_foo_bar:tag_test_err_exist_foo_bar"
+	if err := pushImage(fakeDocker, testImageName, testAuth); err == nil {
+		t.Errorf("Unexpect push image : %v, want error", err)
+	}
```
Comment (bparees) [inline pkg/build/builder/dockerutil_test.go]: this test isn't actually confirming that the error was retried.

File: pkg/build/builder/dockerutil_test.go [hunk 2]
```diff
@@ -108,3 +137,50 @@ func TestTagImage(t *testing.T) {
...
+
+	//expect fail
+	testImageName = "repo_foo_bar:tag_test_err_exist_foo_bar"
+	err := pushImage(fakeDocker, testImageName, testAuth)
+	if err == nil {
+		t.Errorf("Unexpect push image : %v, want error", err)
+	}
+	//expect fail, run 3 times, retry 2 times
+	errMsg := fmt.Sprintf("%s", err)
+	if errMsg != RetriableErrors[2] {
```
Comment (bparees) [inline pkg/build/builder/dockerutil_test.go]: why not just check the value of fooBarRunTimes?

File: pkg/build/builder/dockerutil_test.go [hunk 3]
```diff
@@ -25,6 +29,31 @@ type methodCall struct {
...
+		case 0:
+			fooBarRunTimes++
+			return errors.New(RetriableErrors[0])
+		case 1:
+			fooBarRunTimes++
+			return errors.New(RetriableErrors[1])
+		case 2:
+			fooBarRunTimes = 0
+			return errors.New(RetriableErrors[2])
+		}
```
Comment (bparees) [inline pkg/build/builder/dockerutil_test.go]: if you just get check the value of fooBarRuntimes, you can get rid of this switch and just make it:
fooBarRunTimes++
return errors.New(RetriableErrors[0])

File: pkg/build/builder/dockerutil_test.go [hunk 4]
```diff
@@ -1,12 +1,16 @@
...
 import (
+	"errors"
+	"fmt"
 	"reflect"
 	"testing"
 
 	"github.com/fsouza/go-dockerclient"
 )
 
+var fooBarRunTimes = 0
```
Comment (bparees) [inline pkg/build/builder/dockerutil_test.go]: move this down to be adjacent to the function that's using it.

**Comment by guangxuli** [comment]
@bparees agreed, I had make some modifications, PTAL, thanks

**Comment by bparees** [comment]
lgtm for post 3.3 merge.

**Comment by bparees** [comment]
[merge]

**Comment by guangxuli** [comment]
@bparees Can you take a look for this PR, the test failure looks like not relative to this PR. thanks.

**Comment by bparees** [comment]
@guangxuli yeah we've got some problems w/ the merge queue.  Also this PR is still in the merge queue despite the test failure.

in any case once the merge queue is more stable i'll be revisiting PRs that are ready for merge.

**Comment by bparees** [comment]
flake https://github.com/openshift/origin/issues/11016
[merge]

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
flake https://github.com/openshift/origin/issues/11016
[merge]

**Comment by bparees** [comment]
flake #11016 
[merge]

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
[merge]

**Pull request merged**